Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Commands/Commands.cs

```diff
@@ -180,19 +171,10 @@ public static Command DeleteBinding(int actionMapIndex, int bindingIndex)
                 var binding = Selectors.GetCompositeOrBindingInMap(actionMap, bindingIndex).wrappedProperty;
                 InputActionSerializationHelpers.DeleteBinding(binding, actionMap);
                 state.serializedObject.ApplyModifiedProperties();
-                if (state.selectedBindingIndex >= bindingIndex)
-                    return SelectPrevBinding(state, actionMap);
-                return state.SelectBinding(state.selectedBindingIndex);
-            };
-        }
 
-        private static InputActionsEditorState SelectPrevBinding(InputActionsEditorState state, SerializedProperty actionMap)
-        {
-            var count = Selectors.GetBindingCount(actionMap);
-            var index = -1;
-            if (count != null && count.Value > 0)
-                index = Math.Max(state.selectedBindingIndex - 1, 0);
-            return state.SelectBinding(index);
+                // ActionsTreeView will dispatch a separate command to select the previous Binding
+                return state.With(selectedBindingIndex: -1);
```

Comment:
How about just returning the state without changing the binding index? It will be overwritten by the ActionsTreeView anyway.

Reply:
I was thinking the same thing but unsure about possible side-effects, and so thanks for suggesting this.
Fixed with fbacf6f958456ecc949d89a22634f77574cfc607.